src/ReactPlaceholder.js

```diff
@@ -40,14 +40,20 @@ export default class ReactPlaceholder extends React.Component {
       type, customPlaceholder, showLoadingAnimation, ...rest
     } = this.props;
 
-    if (customPlaceholder) {
-      return customPlaceholder;
-    }
-
     const classes = showLoadingAnimation ?
       ['show-loading-animation', className].filter(c => c).join(' ') :
       className;
 
+    if (customPlaceholder && React.isValidElement(customPlaceholder)) {
+      const mergedCustomClasses =
+        [customPlaceholder.props.className, classes].filter(c => c).join(' ');
+      return React.cloneElement(customPlaceholder,
+        { className: mergedCustomClasses });
+    } else if (!React.isValidElement()) {
```

Comment:
`!React.isValidElement()` always returns `true` as you're checking if `undefined` is a valid react element (and it's not).

You can completely remove this `else if` as the type of `customPlaceholder` is already check by the `propTypes`